stellar/utils.ts

```diff
@@ -186,34 +192,56 @@ async function sendTransaction(tx, server, action, options: Options = {}) {
     }
 }
 
-async function broadcast(operation, wallet, chain, action, options: Options, simulateTransaction = false) {
+async function broadcast(
+    operation: xdr.Operation,
+    wallet: Keypair,
+    chain,
+    action: string,
+    options: Options,
+    simulateTransaction?: boolean,
+): Promise<xdr.ScVal | rpc.Api.SimulateTransactionSuccessResponse> {
+    /* Allows user to require or disallow simulation. Will always attempt if not specified. */
+    if (simulateTransaction === undefined) simulateTransaction = true;
+
     const server = new rpc.Server(chain.rpc, { allowHttp: chain.networkType === 'local' });
+    const tx = await buildTransaction(operation, server, wallet, chain.networkType, options);
+
+    if (simulateTransaction) {
+        const response = await server.simulateTransaction(tx);
+        if (isReadOnly(response)) {
```

Comment:
why the check? Can't we simulate write txs too if we wanted? maybe we're interested in knowing the return value or gas costs etc.?